File: 01 - Historian Hysteria/php/solveA.php
<?php

namespace AoC\One;

use AoC\Common\InputLoader;
use AoC\Common\Logger;

require_once __DIR__ . '/../../common/php/autoload.php';

$logger = new Logger();

$inputPairs = (new InputLoader(__DIR__))->getAsIntArrays();
$logger->log("InputPairs = " . json_encode($inputPairs));

$listA = [];
$listB = [];
foreach ($inputPairs as [$aItem, $bItem]) {
    $listA[]= $aItem;
    $listB[]= $bItem;
}

$logger->log("List A = " . json_encode($listA));
$logger->log("List B = " . json_encode($listB));


sort($listA);
sort($listB);

if (count($listA) !== count($listB)) {
    throw new \Exception('Mismatch in array lengths');
}

$totalDifference = 0;
for($i = 0; $i < count($listA); $i++) {
    $aItem = $listA[$i];
    $bItem = $listB[$i];
    $diff = abs($aItem - $bItem);
    $logger->log("Difference between $aItem and $bItem = $diff");
    $totalDifference += $diff;
}

echo $totalDifference . "\n";

File: 01 - Historian Hysteria/python/solve_a.py
from loader import input_as_ints

if __name__ == "__main__":
    input = input_as_ints()

    listA = []
    listB = []
    for a, b in input:
        listA.append(a)
        listB.append(b)

    listA.sort()
    listB.sort()

    differences = [abs(a - b) for a, b in zip(listA, listB)]
    print(sum(differences))

